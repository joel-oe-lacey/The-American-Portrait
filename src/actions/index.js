export const loadRegion = (region) => ({
    type: 'LOAD_REGION',
    region
})

export const loadCollection = (collection) => ({
    type: 'LOAD_COLLECTION',
    collection
})

export const loadSubsqCollection = (collection) => ({
    type: 'LOAD_SUBSQ_COLLECTION',
    collection
})