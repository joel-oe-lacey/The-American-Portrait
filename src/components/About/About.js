import React from 'react';
import './About.scss';
import { Link } from 'react-router-dom';

const About = () => {
    return (
        <section className="introduction">
            <section className="intro-center">
                <h1>
                    The American Portrait
                </h1>
                <article>
                    "People are trapped in history, and history is trapped in them" said James Baldwin in his 1953 essay "Stranger in the Village". At the time he spoke of the indivisibility of black culture from collective American culture and ethos as a whole. 

                    A continuation of that allegory, I believe, also speaks to the importance of remaining in touch with our collective history. It, in similar duality, is a part of us all. 
                    
                    The American Portrait looks to reunite us with a visual piece of that history. Utilizing the Harvard Museum API, you can view a timeline of historic photos by state, ten per decade where available. 
                    
                    While incomplete and historically biased, these photos offer us at very least a partial window into some of the lives, events, and adolescent places that would go on to form the nation as we know it. 
                </article>
                <Link to="/timeline">Continue</Link>
            </section>
        </section>
    )
}

export default About;