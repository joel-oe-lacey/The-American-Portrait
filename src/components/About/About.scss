.introduction {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background-color: #202020;
    color: white;
}

.intro-center {
    height: 80%;
    width: 60%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;

    h1 {
        font-size: 2em;
    }

    h2 {
        font-size: 1.5em;
    }
}

.intro-selection {
    height: 5%;
    text-align: center;
    font-size: 1.2em;
    width: 75%;
}

.intro-text {
    align-self: flex-start;

    p {
        margin: 4%;
        font-size: 1.2em;
    }
}

.intro-link {
    text-decoration: none;
    font-size: 2em;
    font-weight: bold;
    color: white;
}