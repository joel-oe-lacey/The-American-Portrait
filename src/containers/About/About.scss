.introduction {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: url(../../assets/flag-background.jpg);
    background-size: cover;
}

.intro-center {
    height: 90%;
    width: 60%;
    padding: 2%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    background-color: #202020;
    color: white;

    h1 {
        font-size: 2rem;
    }

    h2 {
        font-size: 1.5em;
    }
}

.intro-selection {
    height: 6%;
    text-align: center;
    font-size: 1.2em;
    width: 75%;
}

.intro-text {
    align-self: flex-start;

    p {
        margin: 4%;
        font-size: 1em;
    }
}

.intro-link {
    text-decoration: none;
    font-size: 2em;
    font-weight: bold;
    color: white;
}