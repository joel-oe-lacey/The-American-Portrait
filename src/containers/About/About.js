import React, { Component }  from 'react';
import './About.scss';
import { Link } from 'react-router-dom';
import { introText, stateNames } from '../../utils/referenceData';
import { loadRegion } from '../../actions';
import { connect } from 'react-redux';

export class About extends Component {
    constructor() {
        super();
        this.state = {
            region: 'Alabama'
        }
    }

    chooseState = e => {
        this.setState({ region: e.target.value });   
    }

    render() {
        const stateSelectOptions = stateNames.map(state => {
            return (
                <option value={`${state}`}>{`${state}`}</option> 
            )
        })

        const introDisplay = introText.map(paragraph => <p>{`${paragraph}`}</p>)
        return (
            <section className="introduction">
                <section className="intro-center">
                    <h1>
                        The American Portrait
                    </h1>
                    <article className="intro-text">
                        {introDisplay}
                    </article>
                    <select 
                        name="state" 
                        onChange={this.chooseState}
                        className="intro-selection">
                        <option value="" disabled selected>Please choose a state.</option>
                        {stateSelectOptions}
                    </select>
                    <Link 
                        to="/region"
                        className="intro-link"
                        onClick={() => this.props.loadRegionToStore(this.state.region)}
                    >Continue</Link>
                </section>
            </section>
        )
    }
}

export const mapDispatchToProps = dispatch => ({
    loadRegionToStore: (region) => { dispatch(loadRegion(region)) },
});

export default connect(null, mapDispatchToProps)(About)