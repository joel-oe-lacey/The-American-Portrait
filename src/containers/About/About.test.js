import React from 'react';
import { About, mapDispatchToProps } from './About';
import { shallow } from 'enzyme';
import { loadRegion } from '../../actions';

describe('About', () => {
    it('should match a snapshot as expected', () => {
        const wrapper = shallow(<About />)
        
        expect(wrapper).toMatchSnapshot()
    })

    describe('mapDispatchToProps', () => {
        it('call dispatch with the loadRegion action when loadRegionToStore is called', () => {
            const mockDispatch = jest.fn();
            const region = "Colorado";
            const actionToDispatch = loadRegion(region)
            const mappedProps = mapDispatchToProps(mockDispatch)

            mappedProps.loadRegionToStore(region)

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        })
    })
})
