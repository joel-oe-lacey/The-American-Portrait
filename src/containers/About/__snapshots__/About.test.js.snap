// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match a snapshot as expected 1`] = `
<section
  className="introduction"
>
  <section
    className="intro-center"
  >
    <h1>
      The American Portrait
    </h1>
    <article
      className="intro-text"
    >
      <p
        key="0"
      >
        People are trapped in history, and history is trapped in them" said James Baldwin in his 1953 essay "Stranger in the Village". At the time he spoke of the indivisibility of black culture from the larger collective American ethos.
      </p>
      <p
        key="1"
      >
        A continuation of that allegory, I believe, also speaks to the importance of remaining in touch with our collective history. It, in similar duality, is a part of us all.
      </p>
      <p
        key="2"
      >
        The American Portrait looks to reunite us with a visual piece of that history. Utilizing the Harvard Museum API, you can view a timeline of historic photos by state, ten per decade where available.
      </p>
      <p
        key="3"
      >
        While incomplete and historically biased, these photos offer us at very least a partial window into some of the lives, events, and adolescent places that would go on to form the nation as we know it. 
      </p>
    </article>
    <select
      className="intro-selection"
      defaultValue=""
      name="state"
      onChange={[Function]}
    >
      <option
        disabled={true}
        value=""
      >
        Please choose a state.
      </option>
      <option
        key="0"
        value="Alabama"
      >
        Alabama
      </option>
      <option
        key="1"
        value="Alaska"
      >
        Alaska
      </option>
      <option
        key="2"
        value="Arizona"
      >
        Arizona
      </option>
      <option
        key="3"
        value="Arkansas"
      >
        Arkansas
      </option>
      <option
        key="4"
        value="California"
      >
        California
      </option>
      <option
        key="5"
        value="Colorado"
      >
        Colorado
      </option>
      <option
        key="6"
        value="Connecticut"
      >
        Connecticut
      </option>
      <option
        key="7"
        value="Delaware"
      >
        Delaware
      </option>
      <option
        key="8"
        value="Florida"
      >
        Florida
      </option>
      <option
        key="9"
        value="Georgia"
      >
        Georgia
      </option>
      <option
        key="10"
        value="Hawaii"
      >
        Hawaii
      </option>
      <option
        key="11"
        value="Idaho"
      >
        Idaho
      </option>
      <option
        key="12"
        value="Illinois"
      >
        Illinois
      </option>
      <option
        key="13"
        value="Indiana"
      >
        Indiana
      </option>
      <option
        key="14"
        value="Iowa"
      >
        Iowa
      </option>
      <option
        key="15"
        value="Kansas"
      >
        Kansas
      </option>
      <option
        key="16"
        value="Kentucky"
      >
        Kentucky
      </option>
      <option
        key="17"
        value="Louisiana"
      >
        Louisiana
      </option>
      <option
        key="18"
        value="Maine"
      >
        Maine
      </option>
      <option
        key="19"
        value="Maryland"
      >
        Maryland
      </option>
      <option
        key="20"
        value="Massachusetts"
      >
        Massachusetts
      </option>
      <option
        key="21"
        value="Michigan"
      >
        Michigan
      </option>
      <option
        key="22"
        value="Minnesota"
      >
        Minnesota
      </option>
      <option
        key="23"
        value="Mississippi"
      >
        Mississippi
      </option>
      <option
        key="24"
        value="Missouri"
      >
        Missouri
      </option>
      <option
        key="25"
        value="Montana"
      >
        Montana
      </option>
      <option
        key="26"
        value="Nebraska"
      >
        Nebraska
      </option>
      <option
        key="27"
        value="Nevada"
      >
        Nevada
      </option>
      <option
        key="28"
        value="New Hampshire"
      >
        New Hampshire
      </option>
      <option
        key="29"
        value="New Jersey"
      >
        New Jersey
      </option>
      <option
        key="30"
        value="New Mexico"
      >
        New Mexico
      </option>
      <option
        key="31"
        value="New York"
      >
        New York
      </option>
      <option
        key="32"
        value="North Carolina"
      >
        North Carolina
      </option>
      <option
        key="33"
        value="North Dakota"
      >
        North Dakota
      </option>
      <option
        key="34"
        value="Ohio"
      >
        Ohio
      </option>
      <option
        key="35"
        value="Oklahoma"
      >
        Oklahoma
      </option>
      <option
        key="36"
        value="Oregon"
      >
        Oregon
      </option>
      <option
        key="37"
        value="Pennsylvania"
      >
        Pennsylvania
      </option>
      <option
        key="38"
        value="Rhode Island"
      >
        Rhode Island
      </option>
      <option
        key="39"
        value="South Carolina"
      >
        South Carolina
      </option>
      <option
        key="40"
        value="South Dakota"
      >
        South Dakota
      </option>
      <option
        key="41"
        value="Tennessee"
      >
        Tennessee
      </option>
      <option
        key="42"
        value="Texas"
      >
        Texas
      </option>
      <option
        key="43"
        value="Utah"
      >
        Utah
      </option>
      <option
        key="44"
        value="Vermont"
      >
        Vermont
      </option>
      <option
        key="45"
        value="Virginia"
      >
        Virginia
      </option>
      <option
        key="46"
        value="Washington"
      >
        Washington
      </option>
      <option
        key="47"
        value="West Virginia"
      >
        West Virginia
      </option>
      <option
        key="48"
        value="Wisconsin"
      >
        Wisconsin
      </option>
      <option
        key="49"
        value="Wyoming"
      >
        Wyoming
      </option>
    </select>
    <Link
      className="intro-link"
      onClick={[Function]}
      to="/region"
    >
      Continue
    </Link>
  </section>
</section>
`;
